### PR TITLE
Added exclude filter

### PR DESCRIPTION
**Pull request by danthe96**
See #114.
In action:
[Search box](https://puu.sh/waVlu/bb6ad4c0a5.png)  
[Filter active](https://puu.sh/waVtN/19416694a9.png)  

Saving the property is implemented in the front end, but not supported by the backend yet. All that needs to be done is adding a property `visibilty_exclude_filter`.

### CODE REVIEW

**Comment by timediv** [comment]
'Show' and 'Exclude' is hard to grasp I believe.
How about 'Show only matching X' and 'Exclude matching Y'

File: assets/js/controllers/AppCtrl.coffee [hunk 1]
```diff
@@ -233,6 +237,7 @@ app.controller 'AppCtrl', [
         bestLimit: newValue.bestLimit
         blacklist: newValue.blacklist.map (f) -> f.id
         searchText: newValue.searchText
+        excludeText: newValue.excludeText
```
Comment (timediv) [inline assets/js/controllers/AppCtrl.coffee]: needs to be added to resetExperimentData as well, right?

**Comment by danthe96** [comment]
Yeah, I was wondering about that as well. I tried something similar at first, but that way the text to filter by doesn't fit inside the cell. So we either have to find something shorter or change the width of the search bar. What would you prefer?

**Comment by timediv** [comment]
Also I see the problem that now, as we have a list of filters in the panel, the sizes becomes way too huge for smaller screens. I think we should have some kind of summary in a single line that can be viewed in detail with a click.

**Comment by timediv** [comment]
Concerning the text length problem: Maybe the items could be two-liners?

**Comment by danthe96** [comment]
Really? There is still only one text filter, one exclude filter, and one div with the chips. Doesn't seem to large to me...

**Comment by danthe96** [comment]
[These items?](https://puu.sh/waVlu/bb6ad4c0a5.png) Not sure if that looks ok, but I'll try it out and attach a screenshot.

**Comment by timediv** [comment]
Yes the search list items.

Oh, sorry, it's not a list but two items.
I still think it's too big for my macbook.
Because it pushes all the main content down which is especially bad with the analysis view.
Maybe there is a way to use extra little padding or something and still have it looking good.
But I would at least try to keep the same space usage as before this pull request.

**Comment by timediv** [comment]
Alternatively we could just add all the items matching the filter to the disabled list. That would also give the user more fine grained control. (Which might be what they actually wanted anyways)
But we would have to consult them to make sure.

**Comment by danthe96** [comment]
'Show only matching' is just barely [short enough](https://puu.sh/waZc2/4049f7b5ca.png) 😄  
I've changed the margin on the filters, it should be small enough now. [Screenshot](https://puu.sh/waZc2/4049f7b5ca.png)

**Comment by danthe96** [comment]
Adding them to the disabled list would make it confusing with filters resulting in hundreds of chips... And removing the chips one-by-one would be kind of cumbersome.

**Comment by danthe96** [comment]
@timediv Update? :)

**Comment by timediv** [comment]
What happens when you enter a very long text filter?

**Comment by danthe96** [comment]
Then it truncates, same as before...

**Comment by danthe96** [comment]
Ok, it's fixed now, multiline is supported

**Comment by timediv** [comment]
Wonderful, I guess we need to get backend support then ;-)

**Comment by danthe96** [comment]
Review please. :) https://github.com/KDD-OpenSource/fexum/pull/171

**Comment by danthe96** [comment]
Backend is merged. :) Can I merge this?

**Comment by nikriek** [comment]
Did you also upload the backend fixes?

**Comment by danthe96** [comment]
Yes, it's already merged into the master and passing.

**Comment by nikriek** [comment]
also the duplicated post thing?

**Comment by nikriek** [comment]
Ah yeah, seems to be alright. Then merge ;)

**Pull request merged**